pkg/prometheus/promcfg_test.go

```diff
@@ -6444,3 +6444,500 @@ func TestTracingConfig(t *testing.T) {
 		})
 	}
 }
+
+func TestServiceMonitorScrapeClass(t *testing.T) {
+	// test table for scrape class
+	testCases := []struct {
+		name        string
+		scrapeClass *monitoringv1.ScrapeClass
+		golden      string
+		expectedErr bool
+	}{
+		{
+			name:        "Service Monitor without Scrape Class",
+			golden:      "serviceMonitorWithoutScrapeClass.golden",
+			scrapeClass: nil,
+		},
+		{
+			name:   "Service Monitor Scrape Class With TLS Config",
+			golden: "serviceMonitorScrapeClassTlsConfig.golden",
+			scrapeClass: &monitoringv1.ScrapeClass{
+				Name: "test-tls-scrape-class",
+				TLSConfig: &monitoringv1.TLSConfig{
+					CAFile:   "/etc/prometheus/secrets/ca.crt",
+					CertFile: "/etc/prometheus/secrets/tls.crt",
+					KeyFile:  "/etc/prometheus/secrets/tls.key",
+				},
+			},
+		},
+		{
+			name:   "Service Monitor Scrape Class With Authorization Config",
+			golden: "serviceMonitorScrapeClassAuthorizationConfig.golden",
+			scrapeClass: &monitoringv1.ScrapeClass{
+				Name: "test-authorization-scrape-class",
+				Authorization: &monitoringv1.Authorization{
+					CredentialsFile: "/etc/prometheus/secrets/credentials",
+				},
+			},
+		},
+		{
+			name:   "Service Monitor Default Scrape Class",
+			golden: "serviceMonitorDefaultScrapeClass.golden",
+			scrapeClass: &monitoringv1.ScrapeClass{
+				Name:    "test-default-scrape-class",
+				Default: ptr.To(true),
+				Authorization: &monitoringv1.Authorization{
+					CredentialsFile: "/etc/prometheus/secrets/default",
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		p := defaultPrometheus()
+
+		monitor := monitoringv1.ServiceMonitor{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "testservicemonitor1",
+				Namespace: "default",
+				Labels: map[string]string{
+					"group": "group1",
+				},
+			},
+			Spec: monitoringv1.ServiceMonitorSpec{
+				Selector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"group": "group1",
+					},
+				},
+				Endpoints: []monitoringv1.Endpoint{
+					{
+						Port:     "web",
+						Interval: "30s",
+					},
+				},
+			},
+		}
```

Comment:
We can, but I'm wondering what will be the other use cases for that?
Do you think we will change the other tests that is only using only one service monitor to this function?
Because on this current context I can't see value other than redeuce line of codes on the test func.

Reply:
@ArthurSens changed 😄